packages/big-design/src/components/Checkbox/Checkbox.tsx

```diff
@@ -58,6 +66,14 @@ const RawCheckbox: React.FC<CheckboxProps & PrivateProps> = ({
     warning('label must be either a string or a CheckboxLabel component.');
   }, [disabled, hiddenLabel, id, label, labelId]);
 
+  const link = description && description.link;
+  const text = description && description.text;
+
+  const renderedDescription = useMemo(() => <FormControlDescription link={link}>{text}</FormControlDescription>, [
+    link,
+    text,
+  ]);
```

Comment:
IMO, this is a little weird. We might want `renderedDescription` to return `null` if `description` is invalid, and use the non-destructured deps i.e. `[..., description.link, description.text]`. That will also play well with @deini  proposed change above.

Reply:
I could use the non-destructured way, but `description` might still be undefined

Reply:
I have pushed a change to allow description to be either string or object, but I think I still need to destructure the variables. To avoid further confusions, maybe you could code a snippet of what you have in mind for this?